test/utils.py

```diff
@@ -290,3 +290,54 @@ def pfb_job_status_in_terra(workspace, job_id):
     else:
         print(resp.content)
         resp.raise_for_status()
+
+
+def add_requester_pays_arg_to_url(url, billing_project='drs-billing-project'):
+    endpoint, args = url.split('?', 1)
+    return f'{endpoint}?userProject={billing_project}&{args}'
+
+
+@retry(error_codes={500, 502, 503, 504})
+def import_drs_from_gen3(drs: str) -> requests.Response:
+    """
+    Import the first byte of a DRS URI using gen3.
+
+    Makes two calls, first one to gen3, which returns the link needed to make the second
+    call to the google API and fetch directly from the google bucket.
+    """
+    if drs.startswith('drs://'):
+        drs = drs[len('drs://'):]
+    else:
+        raise ValueError(f'DRS URI is missing the "drs://" schema.  Please specify a DRS URI, not: {drs}')
+    gen3_endpoint = f'https://staging.gen3.biodatacatalyst.nhlbi.nih.gov/user/data/download/{drs}'
```

Comment:
Which integrations resolve DRS URIs through this endpoint?

Reply:
I'm unclear as to which.  I'll find out.

Some context... ultimately, I want to create a Fence test, but that separate test will log in with gen3, produce three tokens (including the refresh token) and then use them to test the guid fetch here.

I felt the guid fetch might deserve to be a separate test on it's own, and to include a test in a separate PR for a guid that I don't have access to since staging seems to have these testing guids.  Maybe that's overkill though, and I'd be glad for some feedback tomorrow.